### ISSUE TITLE
Environment Canada "met" data format has changed

### ISSUE THREAD

**Issue by richardsc** [dankelley/oce#1607]
I am updating my Barrow Strait shiny app, and noticed that `read.met()` no longer works for data downloaded from the Environment Canada website using `download.met()`. ... *sigh* ...

The old files (that `read.met()` recognized) looked as follows:
```
"Station Name","RESOLUTE BAY A"
"Province","NUNAVUT"
"Latitude","74.72"
"Longitude","-94.97"
"Elevation","67.70"
"Climate Identifier","2403497"
"WMO Identifier",""
"TC Identifier","YRB"
"All times are specified in Local Standard Time (LST). Add 1 hour to adjust for Daylight Saving Time where and when it is observed."

"Legend"
"E","Estimated"
"M","Missing"
"NA","Not Available"

"Date/Time","Year","Month","Day","Time","Temp (°C)","Temp Flag","Dew Point Temp (°C)","Dew Point Temp Flag","Rel Hum (%)","Rel Hum Flag","Wind Dir (10s deg)","Wind Dir Flag","Wind Spd (km/h)","Wind Spd Flag","Visibility (km)","Visibility Flag","Stn Press (kPa)","Stn Press Flag","Hmdx","Hmdx Flag","Wind Chill","Wind Chill Flag","Weather"
"2017-01-01 00:00","2017","01","01","00:00","-18.0","","-21.4","","75","","4","","30","","","M","99.91","","","","-30","","NA"
"2017-01-01 01:00","2017","01","01","01:00","-16.8","","-20.3","","75","","5","","47","","","M","99.90","","","","-31","","NA"
...
```

and the "new" ones look like:
```
"Longitude (x)","Latitude (y)","Station Name","Climate ID","Date/Time","Year","Month","Day","Time","Temp (°C)","Temp Flag","Dew Point Temp (°C)","Dew Point Temp Flag","Rel Hum (%)","Rel Hum Flag","Wind Dir (10s deg)","Wind Dir Flag","Wind Spd (km/h)","Wind Spd Flag","Visibility (km)","Visibility Flag","Stn Press (kPa)","Stn Press Flag","Hmdx","Hmdx Flag","Wind Chill","Wind Chill Flag","Weather"
"-94.97","74.72","RESOLUTE BAY A","2403497","2019-08-01 00:00","2019","08","01","00:00","5.5","","3.6","","88","","9","","39","","","M","99.90","","","","","","NA"
"-94.97","74.72","RESOLUTE BAY A","2403497","2019-08-01 01:00","2019","08","01","01:00","5.6","","4.0","","90","","11","","42","","","M","99.86","","","","","","NA"
...
```

The main difference appears to be that there is no header to the file anymore, but that the primary metadata (lon, lat, station name, climate ID, etc) are all repeated as columns in the table.

Should be an easy fix, but it probably makes sense to code so that the code recognizes and handles both ones, so that it will still work for `met` files downloaded in the past.

Relatedly, I remember sending @dankelley a link for a nice looking R package that could read the data from EC recently ... not sure what it was though. We *could* just hang off another package for reading data.

**Comment by richardsc** [comment]
Looks like they calved off the "metadata" portion to a separate file:
https://climate.weather.gc.ca/climate_data/hourly_data_e.html?hlyRange=2016-01-13%7C2019-10-10&dlyRange=2018-10-29%7C2019-10-10&mlyRange=%7C&StationID=54199&Prov=NU&urlExtension=_e.html&searchType=stnName&optLimit=yearRange&StartYear=1840&EndYear=2019&selRowPerPage=25&Line=0&searchMethod=contains&Month=10&Day=10&txtStationName=resolute&timeframe=1&Year=2019
```
Station Name		RESOLUTE BAY A
Province		NUNAVUT
Current Station Operator	NAV Canada
Latitude		74.72
Longitude		-94.97
Elevation		67.70
Climate Identifier	2403497
WMO Identifier		
TC Identifier		YRB
All times are specified in Local Standard Time (LST). Add 1 hour to adjust for Daylight Saving Time where and when it is observed.

Legend
E	Estimated
M	Missing
NA	Not Available
```

**Comment by richardsc** [comment]
I think [this](https://cran.rstudio.com/web/packages/weathercan/) is the package I remember:

https://cran.rstudio.com/web/packages/weathercan/

**Comment by dankelley** [comment]
I'll have a look.  Unless the fix is complicated - which I doubt - I basically prefer for oce to handle the new format, so users' old code will still work.

The other reason I'd prefer to make small fixes within oce is that having oce work well with other packages is tricky.  Of course, if the user is going to handle the interworking, that's fine (well, so long as we have an `as.` function).  But actual interworking is tricky.  I spent a lot of time recently trying to handle making oce call something from another package. There's so many things to twiddle: do we use `Depends` or `Suggests` in the `DESCRIPTION` file; do we name it in the `NAMESPACE` file; do we use `requireNamespace()` or `require()` - or neither - in the code and the docs; etc.  And the twiddling seems to change over time as R evolves.  And even when the interworking is set up, changes in the other package necessitate changes in oce, so if the data format changes, then users have to wait for the other package to change and then maybe for oce to change in response.

Writing this comment might take longer than writing the code 😃

**Comment by dankelley** [comment]
Hm, the xml file has all the information, i.e. header and data both.

**Comment by dankelley** [comment]
This should work now, in "develop" commit commit 99f4ec8d1fe03cd249711fa27685b7bef7296377

We now have two file fragments in the `extdata`, so users can see what the two files look like.

I'll leave it to @richardsc to decide whether to close this.

I toyed with the idea of also supporting the two-file format but that's a hassle for users, making them download two files like that.  I guess we could add support for the `xml` file format, which has all the data in one file (like the old format...) but I'd prefer to wait until a user needs it.

PS. I removed the `Suggests:` for `marmap`, because we were only using it to get cross-links to documentation, which is a low priority, and because I cannot install `marmap` in R-devel on my macos (catalina) box.
